pkg/apis/apps/validation/validation_test.go

```diff
@@ -127,11 +129,42 @@ func TestValidateStatefulSet(t *testing.T) {
 					}()},
 			},
 		},
+		"maxUnavailable with parallel pod management": {
+			ObjectMeta: metav1.ObjectMeta{Name: "abc-123", Namespace: metav1.NamespaceDefault},
+			Spec: apps.StatefulSetSpec{
+				PodManagementPolicy: apps.ParallelPodManagement,
+				Selector:            &metav1.LabelSelector{MatchLabels: validLabels},
+				Template:            validPodTemplate.Template,
+				Replicas:            3,
+				UpdateStrategy: apps.StatefulSetUpdateStrategy{
+					Type: apps.RollingUpdateStatefulSetStrategyType,
+					RollingUpdate: &apps.RollingUpdateStatefulSetStrategy{Partition: 2, MaxUnavailable: func() *intstr.IntOrString {
```

Comment:
nit: split in different lines

Reply:
+1

Reply:
Also please declare the function above and just call it here? Defining it inline just makes it hard to read.

Also don't return the same pointer every time, if something modifies the thing pointed at it invalidates the other cases and possibly the test comparisons.